app/views/shared/_navbar2.html.erb

```diff
@@ -79,7 +79,11 @@
               </ul>
             </li>
             <%# /* -------------------------------- bookings -------------------------------- */ %>
-            <li> <%= link_to '<i class="fas fa-solid fa-book fa-sm me-1"></i> Bookings'.html_safe,  bookings_path , class: "sub-menu-item" %></li>
+            <% unless current_user.nap_spaces.map(&:bookings).flatten.select(&:requested?).empty? %>
```

Comment:
Looks like a lot of processing in the frontend, let's make a User instance method to return this 😊